src/blockchain/rpc.rs

```diff
@@ -0,0 +1,518 @@
+        wallet_txs.sort_by(|a, b| b.info.confirmations.cmp(&a.info.confirmations));
+
+        // Map wallet transactions to full transactions (TODO: batching)
+        let mut txs: Vec<Transaction> = vec![];
+        for wallet_tx in &wallet_txs {
+            let tx = self
+                .0
+                .as_mut()
+                .ok_or(Error::OfflineClient)?
+                .client
+                .get_transaction(&wallet_tx.info.txid, Some(true))?
+                .transaction()?;
+            txs.push(tx);
+        }
+
+        // Map wallet transactions to block heights (TODO: batching)
+        let mut heights: Vec<Option<u32>> = vec![];
+        for wallet_tx in &wallet_txs {
+            let height = match wallet_tx.info.blockhash {
+                Some(blockhash) => Some(
+                    self.0
+                        .as_mut()
+                        .ok_or(Error::OfflineClient)?
+                        .client
+                        .get_block_info(&blockhash)?
+                        .height as u32,
+                ),
+                None => None,
+            };
+            heights.push(height);
+        }
+
+        Ok(txs
+            .iter()
+            .zip(heights.iter())
+            .map(|(tx, height)| (tx.to_owned(), height.to_owned())) // FIXME
+            .collect())
+    }
+}
+
+impl Blockchain for RpcBlockchain {
+    fn offline() -> Self {
+        RpcBlockchain(None)
+    }
+
+    fn is_online(&self) -> bool {
+        self.0.is_some()
+    }
+}
+
+impl OnlineBlockchain for RpcBlockchain {
+    fn get_capabilities(&self) -> HashSet<Capability> {
+        vec![].into_iter().collect()
+    }
+
+    fn setup<D: BatchDatabase + DatabaseUtils, P: Progress>(
+        &mut self,
+        _stop_gap: Option<usize>,
+        _database: &mut D,
+        _progress_update: P,
+    ) -> Result<(), Error> {
+        let config = self.0.as_mut().ok_or(Error::OfflineClient)?;
+
+        // Check we support their node (rust-bitcoincore-rpc supports 0.18.0 and up)
+        let version = config.client.version()?;
+        if version < 180000 {
+            return Err(Error::BitcoinRpcUnsupportedVersion);
+        }
+
+        // Attempt to load watch-only wallet
+        // FIXME: use listwallets once rust-bitcoincore-rpc supports it
+        match config.client.load_wallet(&config.wallet_name) {
+            Ok(_) => info!("Loaded watch-only wallet: \"{}\"", &config.wallet_name),
+            Err(load_wallet_err) => {
+                // Return if watch-only wallet already exists
+                if let RpcError::JsonRpc(jsonrpc::error::Error::Rpc(ref load_wallet_err)) =
+                    load_wallet_err
+                {
+                    if load_wallet_err.message == format!("Wallet file verification failed: Error loading wallet {}. Duplicate -wallet filename specified.", config.wallet_name) {
+                            info!("Watch-only wallet already loaded: \"{}\"", &config.wallet_name);
```

Comment:
Once [listwallets support is added](https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/121) to rust-bitcoincore-rpc, then we can remove this ugly hack!

Reply:
Maybe here you could use the raw [`call()`](https://docs.rs/bitcoincore-rpc/0.11.0/bitcoincore_rpc/trait.RpcApi.html#tymethod.call) method to make the `listwallets` call. Still kinda hacky but less than having to try I guess